@here/olp-sdk-core/lib/utils/MultiPartUpload.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+import {
+    DataStoreRequestBuilder,
+    HRN,
+    OlpClientSettings,
+    RequestFactory
+} from "@here/olp-sdk-core";
+import { BlobApi, ObjectStoreApi } from "@here/olp-sdk-dataservice-api";
+
+interface MultiPartUploadParams {
+    /**
+     * The instance of OlpClientSettings.
+     */
+    settings: OlpClientSettings;
+    /**
+     * The version od Blob API to upload.
+     */
+    blobVersion: "v1" | "v2";
+    /**
+     * The Layer ID to upload.
+     */
+    layerId: string;
+    /**
+     * If you're uploading to Blob V1, set the datahandle
+     * and set the key for Blob V2.
+     */
+    keyOrDatahandle: string;
+    /**
+     * The Catalog HRN to upload.
+     */
+    catalogHrn: HRN;
+    /**
+     * The content type of a uploaded data.
+     * @see https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.17
+     */
+    contentType: string;
+    /**
+     * The Content Encoding of the data
+     * @note Optional
+     */
+    contentEncoding?: string;
+}
+
+interface UploadParams {
+    file: File | string;
```

Comment:
Why File|string and not ByteArray or any other class used to stream files?
The issue with restricting this to File/path only means that we cannot use it for any other upload where the data is already in RAM and not from a file. E.g. 400MB data that is not saved to a file. This way the user needs to first save the data to file and then call this wrapper. I don't think this is agreeable.

The way to go is to have a common input that can be used either from direct input or from file. In C++ this is std::ostream which can be overloaded for different purposes.

Reply:
Good point. Will update.
The input can be: 
1) File (uploaded from local folder in browser by "Open File" dialog window  (https://developer.mozilla.org/en-US/docs/Web/API/File), can be used only in Browser.
2) string - path to the file, can be used only in NodeJS to read file with stream
3) BufferLike (ArrayBuffer, Buffer) data - to upload data from RAM